README.md

```diff
@@ -1,39 +1,60 @@
+    - `convert/`: test data
+    - `data/`: where you can place your data
+    - `output_charts/`: where the output html and pdf will be saved (see output examples here)
+- `allo_diagram.drawio` is a diagram of the allotaxonometer ecosystem.
+- `raw_requirements.txt` contains the raw requirements for the project which will eventually be used for packaging.
+- Once you set up your ecosystem, you will see `node_modules/`, which will contain the `npm` packages.
+
+
 
 ## Installation steps
 
-1. If environment install is needed (never used or installed `npm`, `nvm`, `node`):
-    - Here are the recommended [steps to install `nvm`](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating). `nvm` is a node version manager that streamlines installing the other 2.
-    - If you have `nvm`, you can do:
+1. If JavaScript tool installs are needed (never used or installed `npm`, `nvm`, `node`):
+    1. Here are the recommended [steps to install `nvm`](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating). `nvm` is a node version manager that streamlines installing the other
+
+    2. Once you have `nvm`, you can run this line to install the latest of both:
         ```shell
         nvm install --lts
         ```
-    to install the latest of both.
-    - Otherwise: [steps to install `node` and `npm`](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
-1. One package depends on having Chrome (brower) or Chromium (browser driver) installed. If you have Chrome installed, you can skip this step. Otherwise, install Chrome or Chromium.
+    - Otherwise (not recommended): [steps to install individually install `node` and `npm`](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+1. One package depends on having Chrome (brower) or Chromium (browser driver) installed. **If you have Chrome installed, you can skip this step**. Otherwise, install Chrome or Chromium.
 Here is the link for the [Chromium download](https://www.chromium.org/getting-involved/download-chromium/).
 
-1. Install the following:
+1. Clone this repo and install the requirements:
     ```shell
     git clone https://github.com/carterwward/py-allotax.git &&
     cd py-allotax &&
-    npm install d3 jsdom jstonge/allotaxonometer#py-allotax-experimental &&
+    npm install @ungap/structured-clone d3 jsdom jstonge/allotaxonometer#py-allotax-experimental &&
     pip install pandas pyhtml2pdf
     ```
-(*note: we are in the process of replacing the HTML->PDF generation method to remove `pyhtml2pdf` dependency*)
+(*note: we will eventually package this repo and streamline installation. See `raw_requirements.txt` for versions and notes if trouble abounds.*)
 
 ## Usage instructions
-1. Verify your data is in the required format by seeing the example in `data/test_data`.
+1. Verify your data is in the required format (`.json`) by seeing examples in `convert/test_data`.
```

Comment:
we should add test for that, assuming users will fail to do it right.

Reply:
I added a function in `utils`